rend3-gltf/src/lib.rs

```diff
@@ -607,6 +745,76 @@ fn load_skins<E: std::error::Error + 'static>(
     Ok(res_skins)
 }
 
+fn load_animations<E: std::error::Error + 'static>(
+    animations: gltf::iter::Animations,
+    buffers: &[Vec<u8>],
+) -> Result<Vec<Labeled<Animation>>, GltfLoadError<E>> {
+    let mut result = Vec::new();
+    for anim in animations {
+        let mut result_channels: Vec<PosRotScale> = Vec::new();
+
+        for (ch_idx, ch) in anim.channels().enumerate() {
+            let target = ch.target();
+            let node_idx = target.node().index();
+
+            // Get the PosRotScale for the current target node or create a new
+            // one if it doesn't exist.
+            let chs = if let Some(chs) = result_channels.iter_mut().find(|chs| chs.node_idx == node_idx as u32) {
```

Comment:
The linearity of this concerns me, if you think the channel count will be low this is fine.

Reply:
Originally, I avoided that because I didn't want to create a HashMap which was then `collect`ed into a `Vec`. But think I'll just make it a `HashMap` and store it in the `Animation` struct as-is.

Originally I wanted to have a vector where position `i` corresponds to the `i`-th joint (just as found in the joint-indices), which is a nice way to store an "associative" structure. But that is not possible in gltf, because there's no guarantee the joint indices will match the node indices, and even if we could reorder them, an animation may be the target of two different skins sharing different subsets of bones... So anyway, HashMap it is. :smile:

Reply:
Thanks for the explaination! Makes sense!